### ISSUE TITLE
Config Hub: use configobj.get_extra_values function to remove deprecated keys in one go

### ISSUE THREAD

**Issue by josephsl** [josephsl/stationPlaylist#95]
Hi,

Another reason for requiring NVDA 2018.4 besides checkable list boxes:

Due to older version of Configobj in use, older NVDA releases lacked a way to remove deprecated keys from config objects. This meant the add-on had to define a list of deprecated keys (in section/key format) that must be removed after caching the normal profile (parsing had to be done first). In NVDA 2018.4 and later (and consequently, in Configobj 4.7.0 and later, with NvDA using 5.1.0 development build), a new function has been added to simplify deprecated keys removal.

The procedure between old and new routines are as follows:

Old:

1. Have a static list of deprecated keys in section/key format.
2. Parse entries in the static deprecated keys list for access to actual keys inside config objects.
3. Check if the old key still exists, and if yes, remove it.

New:

1. ConfigObj.get_extra_values key will return a list of 2-tuples representing extra values for a given config object.
2. Go through these and remove keys unconditionally.

Thus, the following changes will be made:

1. The list of deprecated keys will be removed from SPL config module.
2. Except for some keys, any keys returned by configobj.get_extra_values will be removed. This simplifies the removal loop.

Notes:

1. In case a key moves from global setting to profiles and vice versa, a transitional period must be specified through an extra key.
2. For keys not to be removed, special handling must be done. Currently instant switch profile flag will be eligible for this, and for 19.02, update key will be removed altogether via a different method.

This is destined for stable, and if possible, to LTS18 in a limited capacity. In case this is backported to LTS18, the static list of deprecated keys will be kept as a reminder of old days.

Thanks.

**Comment by josephsl** [comment]
Resolved in 19.03/18.09.7-LTS.
